protocol.html

```diff
@@ -901,6 +901,8 @@ <h3 property="schema:name">Security Considerations</h3>
 
                   <p>Servers <a href="#cors-server">disable all cross-origin protections</a> in browsers because resource access is governed explicitly by the <a href="#authorization">Authorization</a> component. As such, servers cannot rely on browser-based cross-origin protection mechanisms for determining the authentication status or representation of a resource. In particular, servers are strongly encouraged to ignore HTTP cookies from untrusted origins. Additional security measures can be taken to prevent metadata in error responses from leaking. For instance, a malicious application could probe multiple servers to check whether the response status code is <code>401</code> or <code>403</code>, or could try to access an error page from an intranet server within the user agent’s private network to extract company names or other data. To mitigate this, when a request from an untrusted <code>Origin</code> arrives, the server may want to set the status code of error responses to <code>404</code> and/or anonymize or censor their contents.</p>
 
+                  <p>Servers are discouraged from implementing <a href="#storage">multiple storages on the same-origin with untrusted owners</a> because Web apps running in one storage will have the same rights as Web apps running in the other.</p>
```

Comment:
> Because Web apps running in one storage will have the same rights as Web apps running in the other

I think we adding a bit more detail here on the context (e.g. browser-based apps relying on same origin policy) and the attack vectors would be helpful. Understanding the risk better should make it easier for the reader to understand why they are being discouraged from hosting multiple storages in same origin.

Reply:
To continue @justinwb's comment, it is important to be clear about the assumptions that underlie this statement. If, for instance, CORS is used as the basis of identifying applications, which in turn is used for access control, then yes, this would be an important point to make.

If, however, applications are identified through [another mechanism](https://solid.github.io/solid-oidc/#clientids), I struggle to understand why this statement would be needed.

I would also mention that CORS-based restrictions are _only_ going to be relevant for certain browser-based use cases. It is entirely irrelevant for non-browser contexts and even for browser contexts, there are fairly easy ways to circumvent same-origin restrictions (i.e. adding a proxy service)

Reply:
👍 both previous comments

> because Web apps running in one storage

I also find word *running* in this statement possibly leading to some confusion. While application source can be *published* to solid storage, that application *runs* in a web browser on someone's device.

Applications running on remote server (aka 'server side') are perfectly fine, in some ways can be even better, solid applications. Those run on the server at certain origin but currently can't be deployed to solid storage.